src/Symfony/Component/DependencyInjection/Loader/PhpFileLoader.php

```diff
@@ -159,12 +159,16 @@ private function configBuilder(string $namespace): ConfigBuilderInterface
 
         // If it does not start with Symfony\Config\ we dont know how to handle this
         if ('Symfony\\Config\\' !== substr($namespace, 0, 15)) {
-            throw new InvalidargumentException(sprintf('Could not find or generate class "%s".', $namespace));
+            throw new InvalidArgumentException(sprintf('Could not find or generate class "%s".', $namespace));
         }
 
         // Try to get the extension alias
         $alias = Container::underscore(substr($namespace, 15, -6));
 
+        if (false !== strpos($alias, '\\')) {
+            throw new InvalidArgumentException('You can only use "root" ConfigBuilders from "Symfony\\Config\\" namespace. Nested classes are the Plain Old PHP Objects.');
```

Comment:
```suggestion
            throw new InvalidArgumentException('You can only use "root" ConfigBuilders from "Symfony\\Config\\" namespace. Nested classes like "Symfony\\Config\\Framework\\CacheConfig" cannot be used.');
```

Most things are "Plain Old PHP Objects" =)